proxyservice/handlers.go

```diff
@@ -52,16 +52,81 @@ func (d *DockerHubWebhookHandler) ServeHTTP(w http.ResponseWriter, req *http.Req
 		return
 	}
 
-	log.Printf("Triggering Jenkins Job for: %s %s with tag: %s",
-		hookData.Repository.Namespace,
+	rawJSON, err := hookData.rawJSON()
+	if err != nil {
+		log.Printf(err.Error())
+		http.Error(w, "Internal Service Error", http.StatusInternalServerError)
+		return
+	}
+
+	err = d.Jenkins.TriggerJenkinsJob(
+		"dockerhub",
 		hookData.Repository.Name,
+		hookData.Repository.Namespace,
 		hookData.PushData.Tag,
+		rawJSON,
 	)
+	if err != nil {
+		log.Printf("Error triggering jenkins: %v", err)
+		http.Error(w, "Internal Service Error", http.StatusInternalServerError)
+		return
+	}
+
+	w.Write([]byte("OK"))
+}
+
+type GcrWebhookHandler struct {
+	Jenkins      *Jenkins
+	PubSubSecret string
+}
+
+func (d *GcrWebhookHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	if req.Method != "POST" {
+		http.Error(w, "Bad Request", http.StatusBadRequest)
+		return
+	}
+
+	queryParams := req.URL.Query()
 
-	if err := d.Jenkins.TriggerDockerhubJob(hookData); err != nil {
+	if queryParams.Get("secret") != d.PubSubSecret {
```

Comment:
iirc there's a shared secret sent by the gcr service to the proxy in the query params of the url?

Reply:
Yes